lib/src/Uncertainty/Model/CompositeRandomVector.cxx

```diff
@@ -43,9 +44,19 @@ CompositeRandomVector::CompositeRandomVector(const Function & function,
   , function_(function)
   , antecedent_(antecedent)
 {
-  if (function.getInputDimension() != antecedent_.getDimension()) throw InvalidArgumentException(HERE) << "Error: trying to build a CompositeRandomVector from a RandomVector and a Function with incompatible dimensions, here RandomVector dimension=" << antecedent_.getDimension() << " and Function input dimension=" << function.getInputDimension();
+  // compose the function of the inner composite RV
+  CompositeRandomVector * p_composite = dynamic_cast<CompositeRandomVector*>(antecedent.getImplementation().get());
+  if (p_composite && 1)
```

Comment:
A new way to check if a pointer is not null?

Reply:
oh thanks thats was just a debug leftover